backend/src/services/userService.ts

```diff
@@ -112,13 +124,133 @@ const createUser = async (
     }
 };
 
+/**
+ * Fetches and returns list of projects by user id.
+ * returns null on failure
+ */
+const getProjects = async (
+    id: string
+): Promise<{ id: string; name: string }[] | null> => {
+    try {
+        const projects = await selectProjectsbyUserID(id);
+        return projects;
+    } catch (e) {
+        logger.error(e);
+        return null;
+    }
+};
+
+/**
+ * Fetches data of a specific project
+ * along with a boolean value, which is
+ * true if successful or false
+ * if project does not exist or user id
+ * does not match or on failure
+ */
+const getProject = async (
+    projectID: string,
+    userID: string
+): Promise<[boolean, { data: object }]> => {
+    try {
+        const exists = await projectExists(projectID);
+        const ownerID = await selectProjectOwner(projectID);
+        const isOwner = ownerID.userid == userID;
+        if (exists && isOwner) {
```

Comment:
I am fairly certain the exists check is unnecessary here and in the other functions too, as selectProjectOwner will already throw an index out out bounds error if the query does not return any rows. Probably should add a unit test for it.

Reply:
If I'm not wrong, throwing an error would result in a 400 rather than 404, which would go against the specs.